pkg/virtualnode/resource_manager.go

```diff
@@ -311,7 +341,19 @@ func BuildSDKVirtualNodeSpec(vn *appmesh.VirtualNode, vsByKey map[types.Namespac
 		return sdkVSRefConvertFunc(a.(*appmesh.VirtualServiceReference), b.(*string), scope)
 	})
 	sdkVNSpec := &appmeshsdk.VirtualNodeSpec{}
-	if err := converter.Convert(&vn.Spec, sdkVNSpec, nil); err != nil {
+	tempSpec := vn.Spec.DeepCopy()
```

Comment:
dont we get this for free? since existing behavior is that we build the virtual service sdk reference for backends i.e. https://github.com/aws/aws-app-mesh-controller-for-k8s/pull/613/files#diff-93492cb609ae37c6f1dc762777019e79a9eb10d2d20cd6e0c716c776829eafe7R339

Reply:
It does do the conversion, but it does it based on the Backends field of the VirtualNodeSpec, not vsByKey. So we don't get the new backends for free.

Reply:
maybe im missing something but dont we throw all the virtual services from the backend group into the map `vsByKey` and the existing conversion function https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/references/conversions.go#L54 uses that map?

Reply:
Not quite, that conversion function defines conversions for all the vs in vsByKey, but only the backends from the virtual node spec are actually passed as inputs to that function: https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/conversions/virtualnode_types_conversion.go#L647 I've tested without the tempSpec code and the backends from backend groups do not make it into the SDK spec.

Reply:
i see i see reading deeper into the conversion logic i realized the virtual service conversion actually happens here https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/conversions/virtualnode_types_conversion.go#L142 its just registered earlier in this function. i guess the problem i have with adding this logic here is that we have specific functions for converting CRD resources to SDK resources. And in your case i realize you dont have a strict one-to-one SDK resource, but you are converting to SDK backend resources. since the CRD backgroup spec has virtual service references, can we not loop through the virtual services say here https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/conversions/virtualnode_types_conversion.go#L654 and add it to `sdkObj.Backends` ? it also seems like even though we dedupe the virtual service references in the convert func since we use a map, we dont actually do that in the actual conversion since we're appending all backends from backend groups regardless of whether those backends were already provided. i know this is the last thing holding up this review so lmk if you want to chat offline

Reply:
The tempSpec's backends are set to (not appended to) the already de-duped list containing both explicitly defined backends and ones from groups (vsByKey), so there shouldn't be any additional de-duping issue in the conversion step.

I do understand that having the conversion functions effectively ignore backend groups isn't ideal, but moving backendgroup processing into the conversion functions would require passing in a referenceResolver, or at least a k8sClient, in order to get the BackendGroup objects (whose specs contain the VS lists we want to loop through). I don't love this because there's a strong convention that the inputs to those conversion functions are crdObj, sdkObj, scope. Let's go ahead and sync up offline about this.

Reply:
talked offline-- jeremy walked me through what he mentioned above. LGTM